### PR TITLE
document server jvm options

### PR DESCRIPTION
**Pull request by jsanda**
This PR is for #26.

The initial commit only covers a subset of the options for C* 3.11. I want to get some feedback before I do anything more. I am happy to help out with adding some docs however/wherever it is done.

I have always been a fan of Clojure. I even attended the very first Clojure Conj conference, but I definitely do not think user should have to hunt through Clojure code to find keys and their corresponding properties :)

### CODE REVIEW

**Comment by jimdickinson** [comment]
this is awesome! I have a couple specific comments and I would like @cpmcdaniel to check it out before we merge it, but thank you so much for putting this together

File: docs/user/jvm_server_configuration.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+| use_gc_log_file_rotation | `-XX:+UseGCLogFileRotation` | boolean | Disabled by default |
+| `initiating_heap_occupancy_percent` | `-XX:InitiatingHeapOccupancyPercent` | integer | Disabled by default. Can only be used when G1GC garbage collector is used. |
+| `string_table_size` | `-XX:StringTableSize` | string | Defaults to 1000003 |
+| `print_tenuring_distribution` | `-XX:+PrintTenuringDistribution` | boolean | Defaults to false |
+| `cassandra_initial_token` | `-Dcassandra.initial_token` | string | Disabled by default |
+| `resize_tlb` | `-XX:+ResizeTLAB` | boolean | Enabled by default |
+| `cassandra_join_ring` | `-Dcassandra.join_ring` | boolean | Enabled by default |
+| `use_tlb` | `-XX:+UseTLAB` | boolean | Enabled by default |
+| `perf_disable_shared_mem` | `-XX:+PerfDisableSharedMem` | boolean | Enabled by default |
+| `cassandra_config_directory` | `-Dcassandra.config` | string | Disabled by default |
```
Comment (jimdickinson) [inline docs/user/jvm_server_configuration.md:51]: do we leave a note that using this is likely to break everything?

File: docs/user/jvm_server_configuration.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+| `cms_wait_duration` | `-XX:CMSWaitDuration` | integer | Defaults to 10000. Can only be used when CMS garbage collector is used. |
+| `cassandra_replace_address` | `-Dcassandra.replace_address` | string | Disabled by default |
```
Comment (jimdickinson) [inline docs/user/jvm_server_configuration.md:53]: this one too

**Comment by cpmcdaniel** [comment]
I think this is great. Ideally I would like to maybe add some metadata fields in definitions so we can generate this type of documentation. There is already a `:description`, which should satisfy the last column. Perhaps a `:display-name` would be good for the `jvm.options property` column.

**Comment by cpmcdaniel** [comment]
To be clear, that is not a recommendation for this PR. There is a lot of room for improvement on the documentation of config-builder input.

File: docs/user/jvm_server_configuration.md (hunk 1 above)
Comment (jsanda) [reply docs/user/jvm_server_configuration.md:51]: Definitely. I will update it.

File: docs/user/jvm_server_configuration.md (hunk 2 above)
Comment (jsanda) [reply docs/user/jvm_server_configuration.md:53]: Definitely. I will update it.

**Comment by jsanda** [comment]
I have only taken an initial pass through the Cassandra 3.11 options so far. I was thinking about doing a table per server type / version.

I wonder if it would be better to have a single table for Cassandra and just add a column or note to point out properties that are specific to 3.11 or 4.0. Thoughts?

I will follow the same approach for DSE.

**Comment by jsanda** [comment]
I just pushed changes for Cassandra 4.0. 

I am not sure how to document some of the JMX props like:

* `jmx-remote-ssl-opts`
* `jmx-remote-ssl`
* `jmx-remote-ssl-require-client-auth`

I also left out `jmx-port`. Is that another property that should have a warning?

I noticed that there are some [additional properties](https://github.com/datastax/cass-config-definitions/blob/f3f5e40c780100990729abc1c62498c520c26815/resources/jvm11-server-options/cassandra/jvm11-server-options-cassandra-4.0.0.edn) for Java 11. I have not documented these. Should I include them in this PR?

It would great to automate generating these docs. If that is going to happen, does it make sense for me to spend much more time on this?

If I do wind up spending more time on this, then I would prefer to add the DSE sections in a separate PR.

**Comment by jimdickinson** [comment]
Yeah let's see about generating something like this before you spend more time on it. I like the way this looks for now, though!

**Pull request merged**